### PR TITLE
feat: drill by modal

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

- Create a Modal component that takes chart's formData
- Modal title: Drill by: {origin chart's name}
- In the footer, "Close" primary button and "Edit chart" secondary button that does nothing for now

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

drillByModal

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  chartId,
+  formData,
+  initialFilters,
+  showModal,
+  onHideModal,
+}: DrillByModalProps) {
+  console.log(formData);
+  console.log(initialFilters);
+  const theme = useTheme();
+  const { slice_name: chartName } = useSelector(
```
Comment (kgabryje) [inline superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:66]: User can override slice name in dashboard edit mode. Could we use that? Like `state.dashboardLayout.present[chartLayoutItemId].meta.sliceNameOverride || state.dashboardLayout.present[chartLayoutItemId].meta.sliceName`

Comment (lilykuang) [reply superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:66]: is the drill by available in dashboard edit mode?

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx [hunk 2]
```diff
@@ -170,7 +172,9 @@ export const DrillByMenuItems = ({
                 <Menu.Item
                   key={`drill-by-item-${column.column_name}`}
                   {...rest}
-                  onClick={openModal}
+                  onClick={openModal.bind(
+                    column.verbose_name || column.column_name,
```
Comment (kgabryje) [inline superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:176]: Can we pass the whole `column` object to the modal? We need both `verbose_name` (to display in breadcrumbs) and `column_name` (to apply as `groupby`)

**Comment by estebanbouza** [comment]
@lilykuang I'm running this code from the `docker-compose.yml` file in it. I can't see the drill popup when navigating to the same USA Births Names dashboard. It's a clean install (no prior docker images exist in the system, no pre-existing volumes used)
What am I missing to be able to preview this feature?

File: superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:69]: It's not, but user can change chart's displayed name in edit mode. For example, you add chart "My chart" to a dashboard, then in edit mode you rename that chart to "This is my chart" and save. In such case we'd want "This is my chart" to be displayed in drill by modal

**Comment by lilykuang** [comment]
> @lilykuang I'm running this code from the `docker-compose.yml` file in it. I can't see the drill popup when navigating to the same USA Births Names dashboard. It's a clean install (no prior docker images exist in the system, no pre-existing volumes used) What am I missing to be able to preview this feature?

@estebanbouza There is a feature flag `DRILL_BY` need to set to `True` in `superset/config.py`
https://github.com/apache/superset/blob/master/superset/config.py#L481

File: superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}: DrillByModalProps) {
+  const theme = useTheme();
+  const dashboardLayout = useSelector<RootState, DashboardLayout>(
+    state => state.dashboardLayout.present,
+  );
+  const chartLayoutItem = Object.values(dashboardLayout).find(
+    layoutItem => layoutItem.meta?.chartId === formData.slice_id,
+  );
+  const chartName =
+    chartLayoutItem?.meta.sliceName || chartLayoutItem?.meta.sliceNameOverride;
```
Comment (kgabryje) [inline superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:77]: That should be the other way around 😄 sliceNameOverride || sliceName

File: superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  const exploreChart = () => {};
+
```
Comment (kgabryje) [inline superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx:79]: Could you remove console log?

**Review by kgabryje — APPROVED** [review]
Lgtm

**Pull request merged**